src/views.py

```diff
@@ -57,11 +58,45 @@ def get_classrooms(request: HttpRequest) -> HttpResponse:
     for classroom in classrooms:
         building = classroom["building"]
         room = classroom["room"]
-        start_time = ScheduleBlock.objects.filter(room=room).values("start_time")[0]
-        end_time = ScheduleBlock.objects.filter(room=room).values("end_time")[0]
+        day_of_the_week = DAY_OF_THE_WEEK_MAPPING[int(data["day"])]
+
+        start_times = (
+            ScheduleBlock.objects.filter(
+                building=building,
+                room=room,
+                day_of_the_week=day_of_the_week,
+                start_time__gte=data["end_time"],
+            )
+            .order_by("start_time")
+            .values("start_time")
+        )
```

Comment:
This SQL Query does:
1. find all blocks that occur for this building/room/day where the start time of the block occurs after the end time of our search
2. sort by time
3. (later on on line 89 we pull out the first result of this query)

This effectively finds when the classroom is going to be next occupied by a class